src/gui/guiFormSpecMenu.h

```diff
@@ -331,6 +331,13 @@ class GUIFormSpecMenu : public GUIModalMenu
 	u16 m_selected_amount = 0;
 	bool m_selected_dragging = false;
 	ItemStack m_selected_swap;
+	int m_held_button = -1;
+	bool m_shift_craft = false;
```

Comment:
Would you please be so nice to put a short description for each new variable to explain what they're used for?

Reply:
Why? None of the other variables have or need comments.

Reply:
Just because they have none does not make it easier to understand. I am not asking for a detailed explanation. In this case it's not clear what the variable stands for:

 * Does it represent a key state while crafting items?
 * Is it an on/off toggle for some crafting mechanic?

And as for `m_held_button` - is it a keyboard key code or related to mouse? Is it derived from an Irrlicht input enum?

Reply:
I've changed the type and renamed both variables to be more clear: `m_held_mouse_button` and `m_shift_move_after_craft`.

I also had to move the `ButtonEventType` enum into `guiFormSpecMenu.h` to allow using that type there.